zerver/models.py

```diff
@@ -802,24 +804,20 @@ def render_markdown(self, content, domain=None):
             # 'from zerver.lib import bugdown' gives mypy error in python 3 mode.
 
         self.mentions_wildcard = False
-        self.is_me_message = False
         self.mentions_user_ids = set() # type: Set[int]
         self.user_ids_with_alert_words = set() # type: Set[int]
 
         if not domain:
             domain = self.sender.realm.domain
-        if self.sending_client.name == "zephyr_mirror" and domain == "mit.edu":
-            # Use slightly customized Markdown processor for content
-            # delivered via zephyr_mirror
-            domain = u"mit.edu/zephyr_mirror"
+        try:
+            if self.sending_client and self.sending_client.name == "zephyr_mirror" and domain == "mit.edu":
+                # Use slightly customized Markdown processor for content
+                # delivered via zephyr_mirror
+                domain = u"mit.edu/zephyr_mirror"
+        except Client.DoesNotExist:
+            pass
```

Comment:
I need to render markdown before `sending_client` is set. I thought that in this case `self.sending_client` will be just `None` but unfortunately it raises `DoesNotExist`